sync/evmdownloader.go

```diff
@@ -243,59 +247,74 @@ func NewEVMDownloaderImplementation(
 	blockFinality *big.Int,
 	waitForNewBlocksPeriod time.Duration,
 	appender LogAppenderMap,
-	topicsToQuery []common.Hash,
-	adressessToQuery []common.Address,
+	addressesToQuery []common.Address,
 	rh *RetryHandler,
+	finalizedBlockType *big.Int,
 ) *EVMDownloaderImplementation {
 	logger := log.WithFields("syncer", syncerID)
+	var topics []common.Hash
+	if appender != nil {
+		topics = appender.GetTopics()
+	}
+
 	return &EVMDownloaderImplementation{
 		ethClient:              ethClient,
 		blockFinality:          blockFinality,
 		waitForNewBlocksPeriod: waitForNewBlocksPeriod,
 		appender:               appender,
-		topicsToQuery:          topicsToQuery,
-		adressessToQuery:       adressessToQuery,
+		topicsToQuery:          topics,
+		addressesToQuery:       addressesToQuery,
 		rh:                     rh,
 		log:                    logger,
+		finalizedBlockType:     finalizedBlockType,
 	}
 }
 
 func (d *EVMDownloaderImplementation) ChainID(ctx context.Context) (uint64, error) {
 	chainID, err := d.ethClient.ChainID(ctx)
-	if err != nil || chainID == nil {
-		return 0, fmt.Errorf("fail to get chainID from ethClient. Err: %w", err)
+	if err != nil {
+		return 0, fmt.Errorf("failed to retrieve chain id. Err: %w", err)
+	}
+
+	if chainID == nil {
+		return 0, errors.New("chain id is undefined")
 	}
+
 	return chainID.Uint64(), nil
 }
 
 func (d *EVMDownloaderImplementation) GetLastFinalizedBlock(ctx context.Context) (*types.Header, error) {
+	// if the finalized block type is nil, it means that the reorgs are not happening on the network
+	if d.finalizedBlockType == nil {
```

Comment:
I don't think this can be nil. It would fail here:

```
fbtEthermanType := finalizedBlockType
fbt, err := finalizedBlockType.ToBlockNum()
```

Reply:
It certainly can be `nil` (and it would denote that the given network is not prone to reorgs, which should be true for the L2 networks AFAICT):
- https://github.com/agglayer/aggkit/blob/aa160bd90b79c2074252212341cfe547d2598eb3/lastgersync/evmdownloader_fep.go#L50-L53
- https://github.com/agglayer/aggkit/blob/aa160bd90b79c2074252212341cfe547d2598eb3/lastgersync/evmdownloader_pp.go#L59-L62